### PR TITLE
[many ports] Support building with MinGW toolchain

### PR DESCRIPTION
**Pull request by koprok**
Fixes necessary to build some packages with MinGW toolchain in Debian 9. Custom triplet file and CMake toolchain file is not included.

### CODE REVIEW

**Comment by JackBoosY** [comment]
Error in Windows:
```
Included: .\config.msvc
	link.exe /nologo /LIBPATH:\bin.msvc /LIBPATH:C:\vsts\_work\2\s\packages\libxslt_x86-windows\debug\lib /DEBUG /DLL  /VERSION:1.1  /IMPLIB:\bin.msvc\libxslt.lib /OUT:\bin.msvc\libxslt.dll  \int.xslt.msvc\attributes.obj \int.xslt.msvc\documents.obj \int.xslt.msvc\extensions.obj \int.xslt.msvc\extra.obj \int.xslt.msvc\functions.obj \int.xslt.msvc\imports.obj \int.xslt.msvc\keys.obj \int.xslt.msvc\namespaces.obj \int.xslt.msvc\numbers.obj \int.xslt.msvc\pattern.obj \int.xslt.msvc\preproc.obj \int.xslt.msvc\security.obj \int.xslt.msvc\templates.obj \int.xslt.msvc\transform.obj \int.xslt.msvc\variables.obj \int.xslt.msvc\xslt.obj \int.xslt.msvc\xsltlocale.obj \int.xslt.msvc\xsltutils.obj \int.xslt.msvc\attrvt.obj wsock32.lib libiconv.lib libcharset.lib zlibd.lib lzmad.lib libxml2.lib
MSVCRTD.lib(chkstk.obj) : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'ARM'
```

**Comment by koprok** [comment]
> Error in Windows:
> 
> ```
> Included: .\config.msvc
> 	link.exe /nologo /LIBPATH:\bin.msvc /LIBPATH:C:\vsts\_work\2\s\packages\libxslt_x86-windows\debug\lib /DEBUG /DLL  /VERSION:1.1  /IMPLIB:\bin.msvc\libxslt.lib /OUT:\bin.msvc\libxslt.dll  \int.xslt.msvc\attributes.obj \int.xslt.msvc\documents.obj \int.xslt.msvc\extensions.obj \int.xslt.msvc\extra.obj \int.xslt.msvc\functions.obj \int.xslt.msvc\imports.obj \int.xslt.msvc\keys.obj \int.xslt.msvc\namespaces.obj \int.xslt.msvc\numbers.obj \int.xslt.msvc\pattern.obj \int.xslt.msvc\preproc.obj \int.xslt.msvc\security.obj \int.xslt.msvc\templates.obj \int.xslt.msvc\transform.obj \int.xslt.msvc\variables.obj \int.xslt.msvc\xslt.obj \int.xslt.msvc\xsltlocale.obj \int.xslt.msvc\xsltutils.obj \int.xslt.msvc\attrvt.obj wsock32.lib libiconv.lib libcharset.lib zlibd.lib lzmad.lib libxml2.lib
> MSVCRTD.lib(chkstk.obj) : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'ARM'
> ```

@JackBoosY I believe this error has nothing to do with my changes. Maybe I should rebase my branch and then run the checks again?

**Comment by JackBoosY** [comment]
|x86-windows     |test|notes
|----------------|----|-----
|spatialite-tools|Pass|
|readosm         |Pass|
|libxslt         |Fail|**Regression**

@koprok Could you test to install `libxslt:x86-windows` ?

**Comment by koprok** [comment]
@JackBoosY I can install libxslt without any problems. Here is the log from my PC:
```
PS C:\koprok\vcpkg> .\bootstrap-vcpkg.bat                                                                                                                                                               
Building vcpkg.exe ...

  pch.cpp
  archives.cpp
  checks.cpp
  chrono.cpp
  cofffilereader.cpp
  downloads.cpp
  enums.cpp
  files.cpp
  hash.cpp
  machinetype.cpp
  strings.cpp
  stringview.cpp
  system.cpp
  system.print.cpp
  binaryparagraph.cpp
  build.cpp
  commands.autocomplete.cpp
  commands.buildexternal.cpp
  commands.cache.cpp
  commands.ci.cpp
  commands.contact.cpp
  commands.cpp
  commands.create.cpp
  commands.dependinfo.cpp
  commands.edit.cpp
  commands.env.cpp
  commands.exportifw.cpp
  commands.import.cpp
  commands.integrate.cpp
  commands.list.cpp
  commands.owns.cpp
  commands.porthistory.cpp
  commands.portsdiff.cpp
  commands.search.cpp
  commands.upgrade.cpp
  commands.version.cpp
  commands.xvsinstances.cpp
  dependencies.cpp
  export.cpp
  globalstate.cpp
  help.cpp
  input.cpp
  install.cpp
  logicexpression.cpp
  metrics.cpp
  packagespec.cpp
  packagespecparseresult.cpp
  paragraphparseresult.cpp
  paragraphs.cpp
  parse.cpp
  postbuildlint.buildtype.cpp
  postbuildlint.cpp
  remove.cpp
  sourceparagraph.cpp
  statusparagraph.cpp
  statusparagraphs.cpp
  tools.cpp
  triplet.cpp
  update.cpp
  userconfig.cpp
  vcpkgcmdarguments.cpp
  vcpkglib.cpp
  vcpkgpaths.cpp
  versiont.cpp
  visualstudio.cpp
  vcpkglib.vcxproj -> C:\koprok\vcpkg\toolsrc\msbuild.x86.release\vcpkglib.lib
  vcpkg.cpp
  Generating code
  Previous IPDB not found, fall back to full compilation.
  All 14696 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Finished generating code
  vcpkg.vcxproj -> C:\koprok\vcpkg\toolsrc\msbuild.x86.release\vcpkg.exe
  vcpkgmetricsuploader.cpp
  Generating code
  Previous IPDB not found, fall back to full compilation.
  All 5084 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Finished generating code
  vcpkgmetricsuploader.vcxproj -> C:\koprok\vcpkg\toolsrc\msbuild.x86.release\vcpkgmetricsuploader.exe

Building vcpkg.exe... done.

PS C:\koprok\vcpkg> .\vcpkg.exe install libxslt:x86-windows                                                                                                                                             The following packages will be built and installed:
  * libiconv[core]:x86-windows
  * liblzma[core]:x86-windows
  * libxml2[core]:x86-windows
    libxslt[core]:x86-windows
  * zlib[core]:x86-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/5: zlib:x86-windows
Building package zlib[core]:x86-windows...
A suitable version of cmake was not found (required v3.14.0). Downloading portable cmake v3.14.0...
Downloading cmake...
  https://github.com/Kitware/CMake/releases/download/v3.14.0/cmake-3.14.0-win32-x86.zip -> C:\koprok\vcpkg\downloads\cmake-3.14.0-win32-x86.zip
Extracting cmake...
A suitable version of 7zip was not found (required v18.1.0). Downloading portable 7zip v18.1.0...
Downloading 7zip...
  https://www.nuget.org/api/v2/package/7-Zip.CommandLine/18.1.0 -> C:\koprok\vcpkg\downloads\7-zip.commandline.18.1.0.nupkg
Extracting 7zip...
A suitable version of nuget was not found (required v4.6.2). Downloading portable nuget v4.6.2...
Downloading nuget...
  https://dist.nuget.org/win-x86-commandline/v4.6.2/nuget.exe -> C:\koprok\vcpkg\downloads\nuget.exe
A suitable version of powershell-core was not found (required v6.2.1). Downloading portable powershell-core v6.2.1...
Downloading powershell-core...
  https://github.com/PowerShell/PowerShell/releases/download/v6.2.1/PowerShell-6.2.1-win-x86.zip -> C:\koprok\vcpkg\downloads\PowerShell-6.2.1-win-x86.zip
Extracting powershell-core...
A suitable version of git was not found (required v2.21.0). Downloading portable git v2.21.0...
Downloading git...
  https://github.com/git-for-windows/git/releases/download/v2.21.0.windows.1/PortableGit-2.21.0-32-bit.7z.exe -> C:\koprok\vcpkg\downloads\PortableGit-2.21.0-32-bit.7z.exe
Extracting git...
-- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz...
-- Extracting source C:/koprok/vcpkg/downloads/zlib1211.tar.gz
-- Applying patch cmake_dont_build_more_than_needed.patch
-- Using source at C:/koprok/vcpkg/buildtrees/zlib/src/1.2.11-f690224aeb
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Installing: C:/koprok/vcpkg/packages/zlib_x86-windows/share/zlib/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package zlib[core]:x86-windows... done
Installing package zlib[core]:x86-windows...
Installing package zlib[core]:x86-windows... done
Elapsed time for package zlib:x86-windows: 1.335 min
Starting package 2/5: libiconv:x86-windows
Building package libiconv[core]:x86-windows...
-- Downloading https://ftp.gnu.org/gnu/libiconv/libiconv-1.15.tar.gz...
-- Extracting source C:/koprok/vcpkg/downloads/libiconv-1.15.tar.gz
-- Applying patch 0001-Add-export-definitions.patch
-- Applying patch 0002-Config-for-MSVC.patch
-- Using source at C:/koprok/vcpkg/buildtrees/libiconv/src/1.15-a8068c764e
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Installing: C:/koprok/vcpkg/packages/libiconv_x86-windows/share/libiconv/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package libiconv[core]:x86-windows... done
Installing package libiconv[core]:x86-windows...
Installing package libiconv[core]:x86-windows... done
Elapsed time for package libiconv:x86-windows: 11.94 s
Starting package 3/5: liblzma:x86-windows
Building package liblzma[core]:x86-windows...
-- Downloading https://github.com/xz-mirror/xz/archive/v5.2.4.tar.gz...
-- Extracting source C:/koprok/vcpkg/downloads/xz-mirror-xz-v5.2.4.tar.gz
-- Applying patch enable-uwp-builds.patch
-- Using source at C:/koprok/vcpkg/buildtrees/liblzma/src/v5.2.4-66bfe33102
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Installing: C:/koprok/vcpkg/packages/liblzma_x86-windows/share/liblzma/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package liblzma[core]:x86-windows... done
Installing package liblzma[core]:x86-windows...
Installing package liblzma[core]:x86-windows... done
Elapsed time for package liblzma:x86-windows: 20.09 s
Starting package 4/5: libxml2:x86-windows
Building package libxml2[core]:x86-windows...
-- Downloading https://github.com/GNOME/libxml2/archive/v2.9.9.tar.gz...
-- Extracting source C:/koprok/vcpkg/downloads/GNOME-libxml2-v2.9.9.tar.gz
-- Using source at C:/koprok/vcpkg/buildtrees/libxml2/src/v2.9.9-8b0649afae
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Performing post-build validation
-- Performing post-build validation done
Building package libxml2[core]:x86-windows... done
Installing package libxml2[core]:x86-windows...
Installing package libxml2[core]:x86-windows... done
Elapsed time for package libxml2:x86-windows: 22.23 s
Starting package 5/5: libxslt:x86-windows
Building package libxslt[core]:x86-windows...
-- Downloading https://github.com/GNOME/libxslt/archive/v1.1.33.tar.gz...
-- Extracting source C:/koprok/vcpkg/downloads/GNOME-libxslt-v1.1.33.tar.gz
-- Applying patch 0001-Fix-makefile.patch
-- Applying patch 0002-Fix-lzma.patch
-- Using source at C:/koprok/vcpkg/buildtrees/libxslt/src/v1.1.33-8a8517c582
Prerunning x86-windows-dbg
-- Building and installing x86-windows-dbg
Prerunning x86-windows-rel
-- Building and installing x86-windows-rel
-- Warning: Could not find a matching pdb file for:
    C:/koprok/vcpkg/packages/libxslt_x86-windows/bin/libexslt.dll
    C:/koprok/vcpkg/packages/libxslt_x86-windows/bin/libxslt.dll
    C:/koprok/vcpkg/packages/libxslt_x86-windows/debug/bin/libexslt.dll
    C:/koprok/vcpkg/packages/libxslt_x86-windows/debug/bin/libxslt.dll

-- Installing: C:/koprok/vcpkg/packages/libxslt_x86-windows/share/libxslt/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package libxslt[core]:x86-windows... done
Installing package libxslt[core]:x86-windows...
Installing package libxslt[core]:x86-windows... done
Elapsed time for package libxslt:x86-windows: 38.32 s

Total elapsed time: 2.878 min

PS C:\koprok\vcpkg>  

```

**Comment by koprok** [comment]
@JackBoosY, now after this branch builds also on Windows, could you please review it again?

File: ports/glew/portfile.cmake [hunk 1]
```diff
@@ -26,6 +26,7 @@ vcpkg_configure_cmake(
 vcpkg_install_cmake()
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/glew)
 
+if(NOT VCPKG_CMAKE_SYSTEM_NAME OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
```
Comment (JackBoosY) [inline ports/glew/portfile.cmake]: Could you change this condition to `if (VCPKG_TARGET_IS_WINDOWS)`?

File: ports/zlib/portfile.cmake [hunk 2]
```diff
@@ -32,6 +32,7 @@ vcpkg_configure_cmake(
 vcpkg_install_cmake()
 
 # Both dynamic and static are built, so keep only the one needed
+if(NOT VCPKG_CMAKE_SYSTEM_NAME OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
```
Comment (JackBoosY) [inline ports/zlib/portfile.cmake]: Could you change this condition to `if (VCPKG_TARGET_IS_WINDOWS)`?

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Since the port `boost-modular-build-helper` has changed, please correct its version in the file `CONTROL`.
See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

File: ports/ffmpeg/FindFFMPEG.cmake.in [hunk 3]
```diff
@@ -51,7 +51,7 @@ endif()
 #  Platform dependent libraries required by FFMPEG
 if(WIN32)
   if(NOT CYGWIN)
-    list(APPEND FFMPEG_PLATFORM_DEPENDENT_LIBS wsock32 ws2_32 Secur32 bcrypt)
+    list(APPEND FFMPEG_PLATFORM_DEPENDENT_LIBS wsock32 ws2_32 secur32 bcrypt)
```
Comment (JackBoosY) [inline ports/ffmpeg/FindFFMPEG.cmake.in]: Do we need this change? It is not case sensitive in Windows.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/glew/portfile.cmake (hunk 1 above)
Comment (koprok) [reply ports/glew/portfile.cmake]: Done

File: ports/zlib/portfile.cmake (hunk 2 above)
Comment (koprok) [reply ports/zlib/portfile.cmake]: Done

File: ports/ffmpeg/FindFFMPEG.cmake.in (hunk 3 above)
Comment (koprok) [reply ports/ffmpeg/FindFFMPEG.cmake.in]: Yes, if you cross-compile using Linux MinGW toolchain targeting Windows (as I do), it is case-sensitive.

**Comment by koprok** [comment]
@JackBoosY, about the VCPKG_TARGET_IS_WINDOWS changes. I know that MinGW is not officially supported in Vcpkg. But anyway, can you suggest some better way to make difference between Windows/MSVC and Windows/MinGW? When building with MinGW toolchain, the target is still Windows, so this if statement is a little bit misleading, although it works fine in this case.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
When buiding libraqm:x64-windows-static:
```
-- Found FriBidi: C:/vsts/_work/4/s/installed/x64-windows/debug/lib/fribidi.lib  
CMake Error: Could not open file for write in copy operation C:/vsts/_work/4/s/buildtrees/libraqm/src/v0.7.0-bcd3cf2fa3/src/raqm-version.h.tmp
CMake Error: : System Error: Permission denied
CMake Error at CMakeLists.txt:16 (configure_file):
  configure_file Problem configuring file
```

**Comment by koprok** [comment]
> When buiding libraqm:x64-windows-static:
> 
> ```
> -- Found FriBidi: C:/vsts/_work/4/s/installed/x64-windows/debug/lib/fribidi.lib  
> CMake Error: Could not open file for write in copy operation C:/vsts/_work/4/s/buildtrees/libraqm/src/v0.7.0-bcd3cf2fa3/src/raqm-version.h.tmp
> CMake Error: : System Error: Permission denied
> CMake Error at CMakeLists.txt:16 (configure_file):
>   configure_file Problem configuring file
> ```
This is not related to my MinGW changes, but it is easy to fix, so I have pushed a new commit to fix it.

**Comment by koprok** [comment]
@JackBoosY, so everything builds successfully again. Is it anything more I can do to get this merged into master? :-)

**Review by JackBoosY — APPROVED** [review]

File: ports/zlib/portfile.cmake [hunk 4]
```diff
@@ -40,6 +41,16 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
...
 endif()
+else()
+if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
+    if(EXISTS ${CURRENT_PACKAGES_DIR}/lib/libzlibstatic.a)
+        file(RENAME ${CURRENT_PACKAGES_DIR}/lib/libzlibstatic.a ${CURRENT_PACKAGES_DIR}/lib/libzlib.a)
+    endif()
+    if(EXISTS ${CURRENT_PACKAGES_DIR}/debug/lib/libzlibstatic.a)
+        file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/libzlibstatic.a ${CURRENT_PACKAGES_DIR}/debug/lib/libzlibd.a)
+    endif()
+endif()
```
Comment (PhoebeHui) [inline ports/zlib/portfile.cmake]: The library should be renamed through patching the CMake build itself. See [maintainer-guide.md](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-rename-binaries-outside-the-names-given-by-upstream)

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

File: ports/zlib/portfile.cmake (hunk 4 above)
Comment (koprok) [reply ports/zlib/portfile.cmake]: @PhoebeHui, the renaming was already done in the portfile for the MSVC build, so I took the same strategy for the MinGW build. I wanted my changes to touch as less as possible. Do you think it is better to do the renaming for both MSVC and MinGW by patching the zlib CMakeLists.txt? This patch probably would never be accepted upstream. Anyway, I can do it this way if you say so?

**Comment by JackBoosY** [comment]
I found another PR #9137 was solving this issue.

**Comment by koprok** [comment]
@JackBoosY, #9137 is a different thing than my changes.
- #9137 contains:
  - Changes necessary to build **vcpkg itself** with MinGW. This is actually not needed in order to build the ports with MinGW, i.e. you could build vcpkg with MSVC on Windows or GCC on Linux and then use this vcpkg to build the ports targeting MinGW (i.e. cross-compile). This is what I actually do.
  - Adds a **toolchain file** for using MinGW compiler for building the ports. I actually use a similar toolchain file together with a custom triplet file. I intentionally did not add my MinGW toolchain file and triplet file to my commits. I suspected you won't accept these as you don't officially support MinGW yet, right? But anyway, if you merge also #9137 I would be happily using the included MinGW toolchain file.
- My commits are changes to the **ports** that are necessary in order to actually build the ports with MinGW, once after you have the necessary MinGW triplet and toolchain file, be it the one from #9137 or another one.

So, in order to support MinGW both my commits and the ones from #9137 are useful. They are neither both needed nor mutually exclusive.

File: ports/zlib/portfile.cmake (hunk 4 above)
Comment (koprok) [reply ports/zlib/portfile.cmake]: @PhoebeHui, I have patched the zlib CMakeLists file as you have suggested. It's much simpler and cleaner this way.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Can you remove `quantlib:x64-osx=fail` in _VCPKG_PATH/scripts/ci.baseline.txt_?

Please resolve osx build failure:
```
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:25:10: error: non-portable path to file '"tmx.h"'; specified path differs in case from file name on disk [-Werror,-Wnonportable-include-path]
#include "Tmx.h"
         ^~~~~~~
         "tmx.h"
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:32:5: error: use of undeclared identifier 'Tmx'
    Tmx::Map *map = new Tmx::Map();
    ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:32:25: error: use of undeclared identifier 'Tmx'
    Tmx::Map *map = new Tmx::Map();
                        ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:33:10: error: no type named 'string' in namespace 'std'
    std::string fileName = (argc > 1) ? argv[1] : "./example/example.tmx";
    ~~~~~^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:64:16: error: no template named 'unordered_map' in namespace 'std'
    const std::unordered_map<std::string, Tmx::Property> &mapProperties = map->GetProperties().GetPropertyMap();
          ~~~~~^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:64:35: error: no member named 'string' in namespace 'std'
    const std::unordered_map<std::string, Tmx::Property> &mapProperties = map->GetProperties().GetPropertyMap();
                             ~~~~~^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:67:15: error: use of undeclared identifier 'Tmx'
        const Tmx::Property &property = pair.second;
              ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:69:14: error: no type named 'string' in namespace 'std'
        std::string type;
        ~~~~~^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:71:35: error: use of undeclared identifier 'Tmx'
        if (property.GetType() == Tmx::TMX_PROPERTY_STRING)
                                  ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:75:40: error: use of undeclared identifier 'Tmx'
        else if (property.GetType() == Tmx::TMX_PROPERTY_FLOAT)
                                       ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:79:40: error: use of undeclared identifier 'Tmx'
        else if (property.GetType() == Tmx::TMX_PROPERTY_INT)
                                       ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:83:40: error: use of undeclared identifier 'Tmx'
        else if (property.GetType() == Tmx::TMX_PROPERTY_BOOL)
                                       ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:87:40: error: use of undeclared identifier 'Tmx'
        else if (property.GetType() == Tmx::TMX_PROPERTY_COLOR)
                                       ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:91:40: error: use of undeclared identifier 'Tmx'
        else if (property.GetType() == Tmx::TMX_PROPERTY_FILE)
                                       ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:117:12: error: use of undeclared identifier 'Tmx'
    assert(Tmx::Color("#ffffff") == Tmx::Color("#ffffffff"));
           ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:117:37: error: use of undeclared identifier 'Tmx'
    assert(Tmx::Color("#ffffff") == Tmx::Color("#ffffffff"));
                                    ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:128:15: error: use of undeclared identifier 'Tmx'
        const Tmx::Tileset *tileset = map->GetTileset(i);
              ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:145:19: error: use of undeclared identifier 'Tmx'
            const Tmx::Tile *tile = *(tileset->GetTiles().begin());
                  ^
/Users/vagrant/azure-agent/_work/2/s/buildtrees/tmxparser/src/v2.1.0-1a07a7ac14/test/test.cpp:148:40: error: no member named 'string' in namespace 'std'
            std::map<std::string, std::string> list =
                                  ~~~~~^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

**Comment by koprok** [comment]
@JackBoosY, I have made the changes you have requested:
- quantlib removed from baseline
- tmxparser build error on osx should be fixed now
Please, review my changes again.

**Comment by JackBoosY** [comment]
@koprok Please resolve linux regressions.

Thanks.

**Comment by koprok** [comment]
@JackBoosY, all pipelines should be passing now. OS X fails because of some error not related to the code.

**Comment by JackBoosY** [comment]
@koprok I will test it manually.

**Comment by JackBoosY** [comment]
When building fribidi, error shows:
```
  File "/usr/local/bin/meson", line 2
    PYTHONPATH="/usr/local/Cellar/meson/0.53.0/lib/python3.7/site-packages" exec "/usr/local/Cellar/meson/0.53.0/libexec/bin/meson" "$@"
                                                                               ^
SyntaxError: invalid syntax
```

**Comment by koprok** [comment]
> When building fribidi, error shows:
> 
> ```
>   File "/usr/local/bin/meson", line 2
>     PYTHONPATH="/usr/local/Cellar/meson/0.53.0/lib/python3.7/site-packages" exec "/usr/local/Cellar/meson/0.53.0/libexec/bin/meson" "$@"
>                                                                                ^
> SyntaxError: invalid syntax
> ```

@JackBoosY this looks to me like some issue with the build environment that you are using

**Comment by koprok** [comment]
@JackBoosY, @vicroms, all pipelines have passed, finally! Ready to merge now?

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
@koprok Did you test the related features(such as ffmpeg)?

**Comment by koprok** [comment]
@JackBoosY, I have tested only glew. Zlib, bzip2 and ffmpeg are dependencies that are needed by some other libraries that I use. For now it is enough for my project that these can be built successfully with MinGW, but I have not tested them personally. Later maybe I will actually use ffmpeg, but the first step is to be able to build it, right? :)

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Can you remove the following content from _VCPKG_PATH/scirpts/ci.baseline.txt_?
```
kd-soap:x64-linux=fail
jsonnet:x64-osx=fail
```

**Comment by ras0219-msft** [comment]
Awesome work, thanks everyone!

**Pull request merged**